### DISCUSSION TITLE
[Art competition] Grey-box Fuzzing

### DISCUSSION THREAD

**Discussion by jidankim** [prosyslab-classroom/cs348-information-security#419] [Art competition]
![DALL·E 2023-05-31 14 15 29 - View of a faint, not-maintained path (clearly not traveled by people and not groomed; looks dangerous) in the woods on a foggy day, impressionism styl](https://github.com/prosyslab-classroom/cs348-information-security/assets/7444242/9edc5775-73a5-4b8b-a606-64e7e780f2e1)

## Short Description
For me, "Grey-box Fuzzing" means "View of a faint, not-maintained path (clearly not traveled by people and not groomed; looks dangerous) in the woods on a foggy day, impressionism style".

## Long Description (up to 3 sentences)
Grey-box Fuzzing takes a mixture of both Blackbox and Whitebox approaches. Unlike the Blackbox approach, Grey-box requires some guidance toward a goal, usually provided by the experience of previous iterations, as shown by a faint path that had been taken by a few people in this specific trail. Additionally, unlike the Whitebox approach, Grey-box introduces randomness through mutation to explore deeper into the program, like navigating through the woods without a clear map while the path to take is decided by the free will of the current explorer.

**Comment by KihongHeo** [comment]
Good. What kind of feedback you can obtain from such paths?
